transport/src/main/java/io/camunda/zeebe/transport/stream/impl/AggregatedClientStream.java

```diff
@@ -188,11 +182,27 @@ private void tryPush(
             });
   }
 
-  void open(final ClientStreamRequestManager<M> requestManager, final Set<MemberId> servers) {
-    if (state == State.INITIAL) {
-      requestManager.openStream(this, servers);
-      state = State.OPEN;
+  void open(final ClientStreamRequestManager requestManager, final Set<MemberId> servers) {
+    if (isOpened) {
+      return;
+    }
+
+    isOpened = true;
+    for (final var serverId : servers) {
+      requestManager.add(registrationFor(serverId));
+    }
+  }
+
+  ClientStreamRegistration registrationFor(final MemberId serverId) {
+    if (!isOpened) {
+      // TODO: or return optional? anything but null
+      final var registration = new ClientStreamRegistration(streamId, logicalId, serverId);
+      registration.transitionToClosed();
+      return registration;
```

Comment:
Would like to challenge that - I definitely don't want to return null here, so we can return either an `Optional<ClientStreamRegistration>` (which is empty when closed) or an ephemeral, `CLOSED` registration.